# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/HostWebGLContext.h
# Commit: 16d84439756f
# Full Hash: 16d84439756fae6ab67f7e874dd61d03d3b28905
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2020-06-09 09:21:34
# Regressor Bug: 1632249
# File Overlap Count: 1
# Description:
#   Bug 1632249 - WebXR compositing fixes. r=imanol
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D78117
# ==============================================================================

diff -r bbfe23c61add -r 16d84439756f dom/canvas/HostWebGLContext.h
--- a/dom/canvas/HostWebGLContext.h	Mon Jun 08 20:34:15 2020 +0000
+++ b/dom/canvas/HostWebGLContext.h	Mon Jun 08 20:36:14 2020 +0000
@@ -184,9 +184,11 @@
     mContext->SetCompositableHost(compositableHost);
   }
 
-  void Present() { return (void)mContext->Present(); }
-  Maybe<layers::SurfaceDescriptor> GetFrontBuffer(
-      const layers::TextureType t) const;
+  void Present(const ObjectId fb, const layers::TextureType t) const {
+    return (void)mContext->Present(AutoResolve(fb), t);
+  }
+  Maybe<layers::SurfaceDescriptor> GetFrontBuffer(ObjectId fb,
+                                                  layers::TextureType) const;
 
   RefPtr<gfx::DataSourceSurface> GetFrontBufferSnapshot() const {
     return mContext->GetFrontBufferSnapshot();